buddy/buddy/UI/RootCoordinator.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  RootCoordinator.swift
+//  buddy
+//
+//  Created by Laura on 04.03.2023..
+//
+
+import UIKit
+
+final class RootCoordinator: Coordinator {
+    
+    private var childCoordinator: Coordinator?
+    private let persistanceService: PersistenceService
+    
+    init(persistenceService: PersistenceService) {
+        self.persistanceService = persistenceService
+    }
+    
+    func start() -> UIViewController {
+        if persistanceService.isLoggedIn {
+            let mainCoordinator = MainCoordinator()
+            mainCoordinator.parentCoordinator = self
+            childCoordinator = mainCoordinator
+        } else {
+            let authCoordinator = AuthenticationCoordinator()
+            authCoordinator.parentCoordinator = self
+            childCoordinator = authCoordinator
+        }
+        return childCoordinator!.start()
+    }
+    
+    deinit {
+        print("Root coordinator deinitialized")
+    }
+}
+
```

Comment:
```suggestion
// Adding suggestion at bottom of RootController, but please create a separate UIViewController=Extensions file
extension UIViewController {
// Get keyWindow of current application so you can change the rootViewController
    var keyWindow: UIWindow? {
        var window = view.window
        
        if window == nil {
            window = UIApplication.shared.windows.first
        }
        
        return window
    }
    
    // Checks if VC has keyWindow (it will always have it) and sets it rootViewController
    func showAsRoot() {
        if let window = keyWindow {
            window.rootViewController = self
            window.makeKeyAndVisible()
        }
    }
}
```